awscli/bcdoc/restdoc.py

```diff
@@ -102,6 +103,13 @@ def remove_last_doc_string(self):
             start, end = self._last_doc_string
             del self._writes[start:end]
 
+    def write_from_file(self, filename, remove_newlines=False):
+        with open(filename, 'r') as f:
+            for line in f.readlines():
+                if remove_newlines:
+                    line = re.sub('\n', '', line)
```

Comment:
Not ecstatic about this, but global options and synopsis were being reacting differently to being written, so I needed to add this `remove_newlines` flag to create outputs without extra newlines.

Reply:
Yeah I'm not sure if I'm following why we need the `remove_newlines` logic. I would have expected the logic to be just:
```python
def write_from_file(self, filename):
     with open(filename, 'r') as f:
         self.write(f.read())
```

Reply:
Yeah one would definitely think that haha. I tried quite hard to avoid doing this but the reason is because after making the rst files for options and synopsis:
1. `write()` documents them without _any_ newlines eg option1option2option3.
2. `writeln()` documents the two files differently. Global options look fine (single newlines), but global synopsis has double newlines for each option.

I think my problem is that the generated rst files weren't originally intended to be human readable in raw format, so a lot of the code that documents things erred on the side of caution and liberally applied newlines (to avoid problem 1). I couldn't find out where the extra newline was coming from for the global synopsis because these files are pre-made and should just work with `write` but it just doesn't

Reply:
I poked around with this and that rationale seems on track specifically. It looks like `writeln()` is doing a couple of things for us:
1. Adding an extra newline, which guarantees the lines are in a new paragraph when it gets generated into rst and html. 
2. The `writeln` also makes sure we do the appropriate indenting for content. So specifically we need the arguments indented in the synopsis otherwise we generate invalid rst.

I'd say to make this work nicely, let's do the following:

1. Update the helper function to be this:
    ```python
    def write_from_file(self, filename):
        with open(filename, 'r') as f:
            for line in f.readlines():
                self.writeln(line.strip())
    ```
    This will ensure the same number of newlines get added and also guarantee that we have the proper indentation. 

2. Let's go ahead and add the extra newlines in the `global_options.rst` generation logic to add the extra newlines where needed to make sure they are appropriately split up in the help page. So I think we will need to add an extra newline after the `--arg <value>` lines and before the choice enumeration. It makes it less nice to read but this is just intermediary content and the important part is the end content (e.g. html and `help` output).